test/fakes/fakedatastore/fakedatastore.go

```diff
@@ -4,755 +4,258 @@ import (
-		switch req.Mode {
-		case datastore.DeleteBundleRequest_DELETE:
-			for entryID := range bundleEntries {
-				delete(s.registrationEntries, entryID)
-			}
-		case datastore.DeleteBundleRequest_DISSOCIATE:
-			for entryID := range bundleEntries {
-				if entry := s.registrationEntries[entryID]; entry != nil {
-					entry.FederatesWith = removeString(entry.FederatesWith, req.TrustDomainId)
-				}
-			}
-		default:
-			return nil, fmt.Errorf("cannot delete bundle; federated with %d registration entries", len(bundleEntries))
-		}
+	if err := s.getNextError(); err != nil {
+		return nil, err
 	}
-	delete(s.bundles, req.TrustDomainId)
-
-	return &datastore.DeleteBundleResponse{
-		Bundle: cloneBundle(bundle),
-	}, nil
+	return s.ds.DeleteBundle(ctx, req)
 }
 
-// FetchBundle returns the bundle matching the specified Trust Domain.
 func (s *DataStore) FetchBundle(ctx context.Context, req *datastore.FetchBundleRequest) (*datastore.FetchBundleResponse, error) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	if s.expectErr != nil {
-		return nil, s.expectErr
-	}
-
-	bundle, ok := s.bundles[req.TrustDomainId]
-	if !ok {
-		return &datastore.FetchBundleResponse{}, nil
+	if err := s.getNextError(); err != nil {
+		return nil, err
 	}
-
-	return &datastore.FetchBundleResponse{
-		Bundle: cloneBundle(bundle),
-	}, nil
+	return s.ds.FetchBundle(ctx, req)
 }
 
-// ListBundles can be used to fetch all existing bundles.
 func (s *DataStore) ListBundles(ctx context.Context, req *datastore.ListBundlesRequest) (*datastore.ListBundlesResponse, error) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-
-	// get an ordered list of keys so tests can rely on ordering for stability
-	keys := make([]string, 0, len(s.bundles))
-	for key := range s.bundles {
-		keys = append(keys, key)
-	}
-	sort.Strings(keys)
-
-	resp := new(datastore.ListBundlesResponse)
-
-	p := req.Pagination
-	// in case pagination is defined and page size greater than zero apply pagination
-	if p != nil && p.PageSize > 0 {
-		// as default start in first position
-		init := 0
-
-		// If token is defined set index of initial element
-		if p.Token != "" {
-			init = sort.StringSlice(keys).Search(p.Token) + 1
-		}
-
-		// set end as initial element + page size, if end is greater to entries size use length as end
-		length := len(keys)
-		end := init + int(p.PageSize)
-		if end > length {
-			end = length
-		}
-
-		// create a new array with paged bundles
-		keys = keys[init:end]
-		if len(keys) > 0 {
-			lastIndex := len(keys) - 1
-			// change token to latests bundle key
-			resp.Pagination = &datastore.Pagination{
-				PageSize: p.PageSize,
-				Token:    keys[lastIndex],
-			}
-		}
+	if err := s.getNextError(); err != nil {
+		return nil, err
 	}
-
-	for _, key := range keys {
-		resp.Bundles = append(resp.Bundles, cloneBundle(s.bundles[key]))
+	resp, err := s.ds.ListBundles(ctx, req)
+	if err == nil {
+		sort.Slice(resp.Bundles, func(i, j int) bool {
```

Comment:
nit: it may be not super clear why sorting is needed, we may add a comment here.